File: truth_set/go/structure/solution-approach/semantic-dead-code/processCustomOrder.go
package main

import "fmt"

type Item struct {
	Name     string
	Price    float64
	Quantity int
}

type Order struct {
	ID           int
	Items        []Item
	Discount     float64
	CustomerName string
}

func processCustomerOrder(order Order) float64 {
	initialTotal := calculateInitialTotal(order)
	_ = initialTotal    
	var total float64
	for _, item := range order.Items {
		total += item.Price * float64(item.Quantity)
	}
	total += 0 

	logOrderDetails(order) 

	finalTotal := total 
	return finalTotal
}

func calculateInitialTotal(order Order) float64 {
	var sum float64
	for _, item := range order.Items {
		sum += item.Price * float64(item.Quantity)
	}
	return sum
}

func logOrderDetails(order Order) {
}

func main() {
	sampleOrder := Order{
		ID:           123,
		CustomerName: "John Doe",
		Items: []Item{
			{Name: "Laptop", Price: 999.99, Quantity: 1},
			{Name: "Mouse", Price: 49.99, Quantity: 2},
		},
		Discount: 50,
	}

	total := processCustomerOrder(sampleOrder)
	fmt.Printf("Final Total: $%.2f\n", total)
}
\ No newline at end of file

File: src/typescript/structure/solution-approach/semantic-dead-code/processCustomOrder.ts
interface Order {
    id: number;
    items: Array<{ name: string; price: number; quantity: number }>;
    discount?: number;
    customerName: string;
}

function processCustomerOrder(order: Order): number {
    const initialTotal = calculateInitialTotal(order); // This variable is never used.
    let total = 0;
    for (const item of order.items) {
        total += item.price * item.quantity;
    }
    total += 0;

    logOrderDetails(order);

    const finalTotal = total;
    return finalTotal;
}

function calculateInitialTotal(order: Order): number {
    let sum = 0;
    for (const item of order.items) {
        sum += item.price * item.quantity;
    }
    return sum;
}

function logOrderDetails(order: Order): void {
}

const sampleOrder: Order = {
    id: 123,
    customerName: "John Doe",
    items: [
        { name: "Laptop", price: 999.99, quantity: 1 },
        { name: "Mouse", price: 49.99, quantity: 2 },
    ],
    discount: 50,
};

const total = processCustomerOrder(sampleOrder);
console.log(`Final Total: $${total}`);
\ No newline at end of file